azurerm/automation_variable.go

```diff
@@ -0,0 +1,225 @@
+		}
+	} else if value, err = strconv.Unquote(*input); err == nil {
+		actualResource = "azurerm_automation_string_variable"
+	} else if value, err = strconv.ParseBool(*input); err == nil {
+		actualResource = "azurerm_automation_bool_variable"
+	} else if value, err = strconv.ParseInt(*input, 10, 32); err == nil {
+		value = int32(value.(int64))
+		actualResource = "azurerm_automation_int_variable"
+	}
+
+	if actualResource != resource {
+		return nil, fmt.Errorf("Expected value %q to be %q, actual type is %q", *input, resource, actualResource)
+	}
+	return value, nil
+}
+
+func AutomationVariableCommonSchemaFrom(s map[string]*schema.Schema) map[string]*schema.Schema {
+
+	varSchema := map[string]*schema.Schema{
+		"name": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.NoEmptyStrings,
+		},
+
+		"automation_account_name": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.NoEmptyStrings,
+		},
+
+		"description": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+
+		"encrypted": {
+			Type:     schema.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+	}
+	return azure.MergeSchema(s, varSchema)
+}
+
+func resourceArmAutomationVariableCreateUpdate(d *schema.ResourceData, meta interface{}, varType string) error {
+	client := meta.(*ArmClient).automationVariableClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	accountName := d.Get("automation_account_name").(string)
+	varTypeLower := strings.ToLower(varType)
+
+	if requireResourcesToBeImported {
+		resp, err := client.Get(ctx, resourceGroup, accountName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Error checking for present of existing Automation %s Variable %q (Automation Account Name %q / Resource Group %q): %+v", varType, name, accountName, resourceGroup, err)
+			}
+		}
+		if !utils.ResponseWasNotFound(resp.Response) {
+			return tf.ImportAsExistsError(fmt.Sprintf("azurerm_automation_%s_variable", varTypeLower), *resp.ID)
+		}
+	}
+
+	description := d.Get("description").(string)
+	encrypted := d.Get("encrypted").(bool)
+	value := ""
+
+	if varTypeLower == "datetime" {
+		vTime, parseErr := time.Parse(time.RFC3339, d.Get("value").(string))
+		if parseErr != nil {
+			return fmt.Errorf("Error invalid time format: %+v", parseErr)
+		}
+		value = fmt.Sprintf("\"\\/Date(%d)\\/\"", vTime.UnixNano()/1000000)
+	} else if varTypeLower == "bool" {
+		value = strconv.FormatBool(d.Get("value").(bool))
+	} else if varTypeLower == "int" {
+		value = strconv.Itoa(d.Get("value").(int))
+	} else if varTypeLower == "string" {
+		value = strconv.Quote(d.Get("value").(string))
+	}
+
+	parameters := automation.VariableCreateOrUpdateParameters{}
+
+	if varTypeLower == "null" {
```

Comment:
would this work instead:
```go
	parameters = automation.VariableCreateOrUpdateParameters{
		Name: utils.String(name),
		VariableCreateOrUpdateProperties: &automation.VariableCreateOrUpdateProperties{
			Description: utils.String(description),
			IsEncrypted: utils.Bool(encrypted),
		},
	}
	if varTypeLower != "null" {
		parameters.VariableCreateOrUpdateProperties.Value =utils.String(value),
	}
```

Reply:
Yeah, I actually changed that just after I pushed the code... lol